### PR TITLE
CHORE: Move `page.window_*` and `page.browser_context_menu_*` properties to `Window` and `BrowserContextMenu` classes

### LINKED ISSUE #3414: Create new `Window` class for `page.window_*` properties
## Tasks
- [x] Create `Window` class and init in `Page`
- [x] Move `page.window_*` properties to `WIndow`
- [x] Deprecate `page.window_*` properties

## Idea/Goal
```py
# before
page.window_height

# after
page.window.height
```

### PR DESCRIPTION
**Pull request by ndonkoHenri**
## Description

- Moved `page.window_*` and `page.browser_context_menu_*` properties to `Window` and `BrowserContextMenu` classes respectively
- Deprecated `page.window_*` props in favor of `page.window.*`
- Exposed some more properties of the window_manager packages (`shadow`, `alignment`, `wait_until_ready_to_show`, `always_on_bottom`, `icon`, `badge_label`)

Fixes #3414 

## Test Code

```python
import flet as ft


def main(page: ft.Page):
    page.window_always_on_top = True  # deprecated
    page.window.title = "Hello, World!"
    page.window.on_event = lambda e: print(e.type)
    page.on_resize = lambda e: print(e.width, e.height)  # deprecated
    page.window.shadow = True  # new
    page.window.badge_label = "2"  # new (macos only)
    page.window.alignment = ft.alignment.center_right  # new
    page.window.always_on_bottom = False  # new
    page.window.wait_until_ready_to_show = True  # new

    page.add(
        ft.Text("Hello, World!", size=30),
    )


ft.app(target=main)

```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] I signed the CLA.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

- [x] New and existing tests pass locally with my changes
- [x] I have made corresponding changes to the [documentation](https://github.com/flet-dev/website) (if applicable)

## Additional details

`page.window.icon` was added but not tested since it is windows-only (which I don't have)

### CODE REVIEW

File: sdk/python/packages/flet-core/src/flet_core/page.py [hunk 1]
```diff
@@ -126,6 +129,377 @@ def __init__(self, message):
...
+        self.page._invoke_method("disableBrowserContextMenu", wait_timeout=wait_timeout)
+        self.disabled = True
+
+
+class Window:
+    def __init__(self, page: "Page"):
+        self.page = page
+        self.__alignment = None
+        self.__on_event = EventHandler(lambda e: WindowEvent(e))
+        self.__on_resized = EventHandler(lambda e: WindowResizeEvent(e))
```
[the remark this answers is not in the document]
Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/page.py:151]: Nice catch! :)

File: sdk/python/packages/flet-core/src/flet_core/page.py [hunk 2]
```diff
@@ -126,6 +129,377 @@ def __init__(self, message):
...
+    def icon(self, value: Optional[str]):
+        self.page._set_attr("windowIcon", value)
+
+    # Methods
+    def destroy(self):
+        self.page._set_attr("windowDestroy", True)
+        self.page.update()
+
+    def center(self) -> None:
+        self.page._set_attr("windowCenter", str(time.time()))
```
[the remark this answers is not in the document]
Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/page.py:476]: @FeodorFitsner, wdyt?

File: sdk/python/packages/flet-core/src/flet_core/utils/deprecated.py [hunk 3]
```diff
@@ -27,6 +27,15 @@ def new_func(*args, **kwargs):
     return decorator
 
 
+def deprecation_warning(d: str, reason: str, version: str, delete_version: str) -> None:
```
[the remark this answers is not in the document]
Comment (ndonkoHenri) [reply sdk/python/packages/flet-core/src/flet_core/utils/deprecated.py:30]: Nice suggestion!

File: sdk/python/packages/flet-core/src/flet_core/page.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FeodorFitsner) [reply sdk/python/packages/flet-core/src/flet_core/page.py:474]: That makes sense.

**Review by FeodorFitsner — APPROVED** [review]

**Pull request merged**